File: app/src/main/java/antoninovitale/dropcodechallenge/details/section/model/Add.java
package antoninovitale.dropcodechallenge.details.section.model;

/**
 * Created by antoninovitale on 31/08/2017.
 */
public enum Add {
    START, MIDDLE, END, NONE;

    public static Add getAdd(String add) {
        if (add == null) return NONE;

        switch (add) {
            case "start":
                return START;
            case "middle":
                return MIDDLE;
            case "end":
                return END;
            default:
                return NONE;
        }
    }

}

File: app/src/main/java/antoninovitale/punkapi/app/details/section/model/Add.kt
package antoninovitale.punkapi.app.details.section.model

/**
 * Created by antoninovitale on 31/08/2017.
 */
enum class Add {
    START, MIDDLE, END, NONE;

    companion object {
        fun getAdd(add: String?): Add {
            return if (add == null) NONE else when (add) {
                "start" -> START
                "middle" -> MIDDLE
                "end" -> END
                else -> NONE
            }
        }
    }
}